src/screens/NewTerm.js

```diff
@@ -0,0 +1,201 @@
+import React, {useEffect, useState} from 'react';
+import {Picker} from '@react-native-picker/picker';
+import {action} from '@storybook/addon-actions';
+import 'react-native-get-random-values';
+import {v4 as uuid} from 'uuid';
+import {
+  Text,
+  View,
+  StyleSheet,
+  SafeAreaView,
+  KeyboardAvoidingView,
+} from 'react-native';
+
+import ToolBar from '../components/ToolBar/ToolBar';
+import Textarea from '../components/Textarea/Textarea';
+import NextButton from '../components/NextButton/NextButton';
+import ToolButton from '../components/ToolButton/ToolButton';
+import LanguageSwitcher from '../components/LanguageSwitcher/LanguageSwitcher';
+import BackIcon from '../components/ToolButton/assets/back.svg';
+import ModeIcon from '../components/ToolButton/assets/mode.svg';
+
+import {LANGUAGE_NAMES, getAllCategories} from '../data/dataUtils';
+import SectionHeading from '../components/SectionHeading/SectionHeading';
+
+export default ({
+  //nav provider
+  navigation,
+  categories,
+  setCategories,
+  addNewPhrases,
+}) => {
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [malagasyText, setMalagasyText] = useState('');
```

Comment:
If you combine the state for the english and malagasy text in an object like {en: "", mg:""},  you can reuse it in line 47 more easily.